documentation/GETTING_STARTED.md

```diff
@@ -0,0 +1,349 @@
+
+```bash
+# If not already in the `getting_started_guide` directory:
+cd /documentation/getting_started_guide/
+
+python
+```
+
+Paste the following steps into the terminal. (Alternatively, you can also create a Python script, copy in the code below, and run it. The full script can be found as [`process_data.py` in the `scripts` directory](./getting_started_guide/scripts/process_data.py).)
+
+#### 1. Import dependencies and load the dataset into a pandas DataFrame
+
+```python
+from bioio import BioImage
+import pandas as pd
+
+from colorizer_data.types import DATETIME_FORMAT
+from colorizer_data.utils import (
+    remap_segmented_image,
+)
+from colorizer_data.writer import (
+    ColorizerDatasetWriter,
+    ColorizerMetadata,
+    FeatureInfo,
+    FeatureType,
+)
+
+# Load the dataset
+data: pd.DataFrame = pd.read_csv("raw_dataset/data.csv")
+```
+
+#### 2. Configure the writer and data columns
+
+```python
+# Define column names
+OBJECT_ID_COLUMN = "object_id"
+TRACK_ID_COLUMN = "track_id"
+TIMES_COLUMN = "time"
+SEGMENTED_IMAGE_COLUMN = "segmentation_path"
+CENTROIDS_X_COLUMN = "centroid_x"
+CENTROIDS_Y_COLUMN = "centroid_y"
+AREA_COLUMN = "area"
+LOCATION_COLUMN = "location"
+RADIUS_COLUMN = "radius"
+
+# Add in a column to act as an index for the dataset.
+# This preserves row numbers even when the dataframe is grouped by
+# time later.
+INDEX_COLUMN = "index"
+data = data.reset_index(drop=True)
+data[INDEX_COLUMN] = data.index.values
+
+# Create the writer, which we'll be using to save dataset-related files.
+output_dir = "."
+dataset_name = "processed_dataset"
+writer = ColorizerDatasetWriter(output_dir, dataset_name)
+```
+
+#### 3. Write the core data
+
+```python
+# Turn core data columns into a numpy array, to be saved by the writer.
+tracks = data[TRACK_ID_COLUMN].to_numpy()
+times = data[TIMES_COLUMN].to_numpy()
+centroids_x = data[CENTROIDS_X_COLUMN].to_numpy()
+centroids_y = data[CENTROIDS_Y_COLUMN].to_numpy()
+
+writer.write_data(
+    tracks=tracks,
+    times=times,
+    centroids_x=centroids_x,
+    centroids_y=centroids_y,
+)
+```
+
+#### 4. Write the features
+
+Features can be one of three types:
+
+1. **Continuous** features are used for floating-point numbers.
+2. **Discrete** features are used for integers.
+3. **Categorical** features are used for string-based labels. Note that there can only be a max of 12 categories for a categorical feature.
+
+The `FeatureInfo` class is used to provide metadata about each feature, such as its label, key, type, and units if applicable.
+
+```python
+areas = data[AREA_COLUMN].to_numpy()
+locations = data[LOCATION_COLUMN].to_numpy()
+radii = data[RADIUS_COLUMN].to_numpy()
+
+# Additional metadata can be provided for each feature, which will be shown
+# when interacting with it in the viewer.
+area_info = FeatureInfo(
+    label="Area",
+    key="area",
+    type=FeatureType.CONTINUOUS,
+    unit="px²",
+)
```

Comment:
seeing this here makes me think it could be elegant to just put the numpy array in the FeatureInfo class, and then write_feature can be passed just a FeatureInfo alone!  Like `FeatureInfo(data=areas, label="Area" , ...)` something to ponder for later..